doc/BGP/BGP-route-aggregation-with-bbr-awareness.md

```diff
@@ -0,0 +1,349 @@
+
+Secondly, if we aggregated routes without BBR feature on device, we many got packet drop on this device due to contributing routes missing.
+
+To leverage the benefit of address aggregation and the BBR feature, we are trying to design the aggregate address configuration mechanism with BBR awareness in this doc.
+
+
+## Requirements
+With this design, users can add or remove aggregated address via editing config DB or config CLI with parameters to control the aggregation and route announcement behavior and can use the show command to check aggregated address on device.
+
+With BBR awareness, aggregated address will be generated conditionally depend on BBR feature's state.
+
+## High Level Design
+First we introduce the config DB extension which define the feature scope and parameters we have.
+Then we introduce how the bgp container will change its behavior accordingly.
+
+### Config DB Extension
+We define a new YANG model to add a new key named `BGP_AGGREGATE_ADDRESS` in config DB.
+The key will index to a list of aggregated addresses with their parameters.
+
+The YANG model and config DB demo are showed in below:
+
+#### Yang Model
+```
+module sonic-bgp-aggregate-address {
+
+    namespace "http://github.com/sonic-net/sonic-bgp-aggregate-address";
+
+    prefix bgp-aggregate-address;
+
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP aggregate address configuration module.";
+
+    revision 2024-07-17 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-aggregate-address {
+        container BGP_AGGREGATE_ADDRESS {
+
+            description "BGP_AGGREGATE_ADDRESS part of config_db.json";
+
+            list BGP_AGGREGATE_ADDRESS_LIST {
+
+                description "BGP_AGGREGATE_ADDRESS list part of config_db.json";
+
+                key "aggregate-address";
+
+                leaf aggregate-address {
+                    type inet:ip-prefix;
+                    description "Aggregate address to be advertised";
+                }
+
+                leaf bbr-required {
+                    type boolean;
+                    default false;
+                    description "Set if BBR is required for generating aggregate address";
+                }
+
+                leaf summary-only {
+                    type boolean;
+                    default false;
+                    description "Only advertise the summary of aggregate address";
+                }
+
+                leaf as-set {
+                    type boolean;
+                    default false;
+                    description "Set if include the AS set when advertising the aggregated address";
+                }
+
+                leaf aggregate-address-prefix-list {
+                    type string {
+                        pattern "[0-9a-zA-Z_-]*";
```

Comment:
Let's limit length as well: https://github.com/FRRouting/frr/blob/536d478049b4ae7c3d60676a75a56115491d9b6f/yang/frr-filter.yang#L57

FRR does not have too much restriction on the name, but we need to consider abuse. You can make sure no injection either by code or yang.

Reply:
Yes, I will add length restriction from 0 to 128, and 0 works for empty string, I have tested that 0 works on testbed.